src/main/java/org/folio/circulation/services/LostItemFeeChargingService.java

```diff
@@ -174,19 +192,17 @@ private String getFeeFineActionCreatedAt(ReferenceDataContext context) {
 
   private boolean shouldChargeItemFee(LostItemPolicy policy) {
     // Set cost fee is only supported now
-    return policy.getChargeAmountItem().getChargeType() == ChargeAmountType.SET_COST
-      && isGreaterThanZero(policy.getChargeAmountItem().getAmount());
+    return policy.getChargeAmountItem()
+      .filter(chargeAmount -> chargeAmount.getChargeType() == SET_COST)
+      .filter(chargeAmount -> isGreaterThanZero(chargeAmount.getAmount()))
+      .isPresent();
```

Comment:
Unfortunately they can not be reused. 
`setCostItemFeeHasZeroValue` should match policy when: `chargeType` is `SET_COST` AND `amount` <= `0`,
but `shouldChargeItemFee` matches policy when `chargeType` is `SET_COST` AND `amount` > `0`

`!setCostItemFeeHasZeroValue` will also match policies:
* Is not `SET_COST` and has amount < 0;
* Is not `SET_COST` and has amount > 0;

Reply:
I think I'm a little lost. Why is there a need for both checks?

As I understand it, either the amount is greater than zero and a fee is charged to the patron, or it is not and the fee is ignored.

Am I missing some additional logic that is required?

Reply:
We have to take into account fee charge type as well, it can be either `set cost` or `actual cost`, for the last one amount can be unspecified at all, however it should be kept open.

Reply:
I think I'm still a little lost.

Are these the scenarios we are trying to account for?
* Set cost of zero
* Set cost of greater than zero
* Set cost of less than zero
* Actual cost (where the amount should not be set at all)

Reply:
Actual cost is out of the scope. This method is used to verify that policy has **set cost charge** type and amount is  > 0.

The another method `setCostItemFeeHasZeroValue` verifies that policy has **set cost charge** type but amount is <= 0.

What I'm trying to say is that those two methods can not be reused because each of them is not an negation of other one, they can be described in following way:

* **methodA** is defined as => A && B;
* **methodB** is defined as => A && !B;

**!methodA** matches following conditions => !A && !B, !A && B, A && !B, the same for **!methodB**. As you can see, there are undesired matches, so we have to duplicate those two methods.

Reply:
@bohdan-suprun  Thanks for trying to explain this to me. I think I'm still failing to follow it.

I'm going to close this comment and revisit it during the work to model the various costs more explicitly.